src/components/MyObservations/ObsUploadStatus.js

```diff
@@ -33,11 +31,10 @@ const ObsUploadStatus = ( {
   const startSingleUpload = obsEditContext?.startSingleUpload;
   const uploadProgress = obsEditContext?.uploadProgress;
   const wasSynced = observation.wasSynced( );
-  const { allObsToUpload } = uploadStatus;
   const whiteColor = white && theme.colors.onPrimary;
 
   const displayUploadStatus = ( ) => {
-    if ( allObsToUpload?.find( upload => upload.uuid === observation.uuid ) ) {
+    if ( !observation.id ) {
```

Comment:
Unrelated to #571 but there is a moment when `ObsStatus` shows right when you add an unuploaded observation.

I believe unuploaded observations do not have an ID (but have a uuid). So it's faster to check for id truthiness and thus gets rid of the flash of ObsStatus.

However, if this isn't a reliable way to detect if an Observation has been uploaded, then I will revert.